src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentDirectiveIntegrationTest.cs

```diff
@@ -100,6 +100,23 @@ public void SupportsInjectDirective()
             s => AssertEx.Equal("private TestNamespace.IMyService2 Test.TestComponent.MyService2 { get; set; }", s.ToTestDisplayString()));
     }
 
+    [Fact]
+    public void SupportsIncompleteInjectDirectives()
+    {
+        var component = CompileToComponent("""
+            @inject 
```

Comment:
It's not clear to me how this resolves the issue. The issue talks about IntelliSense on the type part and when user writes `@inject` and space we still don't generate anything, i.e., nothing has really changed for this sceario, how could the issue be fixed?

Reply:
I am also confused here.

Reply:
That's always been true. You don't get IntelliSense until you type the first character today. Once you type the first character, you get it, but in fuse you never get it until you start typing the name:

Not-fuse:

![not_fuse](https://github.com/user-attachments/assets/9945f7e4-4160-43b8-b4ec-82b8981f35f1)

Fuse (broken):

![fuse_broken](https://github.com/user-attachments/assets/4148c047-e212-4d3f-8b75-abd9b690cb34)

Fuse (fixed):

![fuse_fixed](https://github.com/user-attachments/assets/7716b3e9-a01b-400b-83ef-16301b075540)


Note, this is same behavior you get today in C#, if you're e.g. writing a new property:

![csharp_property](https://github.com/user-attachments/assets/24886d77-958d-4f95-b217-f8b64b8af280)

Reply:
In C#, can't you manually bring up completion though? What's the equivalent behavior in razor currently?

Reply:
Ah ok, so if you invoke IntelliSense in either the C# or non-fuse razor case, you just get an empty IntelliSense box.
![image](https://github.com/user-attachments/assets/e212fb57-abd3-439b-a81c-3da9592dba3f)

In the PR right now you get nothing, because we have nothing emitted. In design time that works by mapping to just a totally empty space. We can probably do that here too to get exactly the same behavior.

Reply:
Hmm, actually this is proving much harder than expected. Not necessarily in generating the code but trying to figure out what exactly we should generate to make IntelliSense happy. I'll chat with @davidwengier next week and try and figure it out. 

Should we just take this PR as-is as it makes most of the scenario better, and file a follow up issue to fix the empty space part?